implementations/rust/ockam/ockam_api/src/session/util.rs

```diff
@@ -5,19 +5,44 @@ use std::time::Duration;
 pub(crate) const MAX_RECOVERY_TIME: Duration = Duration::from_secs(10);
 pub(crate) const MAX_CONNECT_TIME: Duration = Duration::from_secs(5);
 
-pub(crate) fn starts_with_host_tcp(addr: &MultiAddr) -> Option<usize> {
+pub(crate) fn starts_with_host_tcp(addr: &MultiAddr) -> Option<(MultiAddr, MultiAddr)> {
     let host_match = Match::any([DnsAddr::CODE, Ip4::CODE, Ip6::CODE]);
     if addr.matches(0, &[host_match, Tcp::CODE.into()]) {
-        Some(2)
+        Some(addr.split(1))
```

Comment:
This should still stick with the index of 2.

When we split a TCP, we need both the IP address and Port

An example multi_addr would be`/dnsaddr/localhost/tcp/4000/service/api`, the tcp address is `/dnsaddr/localhost/tcp/4000` (localhost:4000) and the remaining part of the address is `/service/api`